### PR TITLE
fix(OverflowMenu): stop focusing button onblur

### LINKED ISSUE #6227: OverflowMenu steals back focus when clicking outside and it collapses
## What package(s) are you using?

- [ ] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.

If I click on a button (or really anything) while the OverflowMenu is expanded.  It steals focus back to the collapsed Menu button instead of focusing on what I clicked on

> Is this issue related to a specific component?

> What did you expect to happen? What happened instead? What would you like to
> see changed?

> What browser are you working in?

> What version of the Carbon Design System are you using?

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

## Steps to reproduce the issue

Take a look at the sandbox below.
Expand the Overflow Menu
Now click on the button next to the Overflow Menu.
Notice that the OverflowMenu Icon has stolen the focus back to itself

> Please create a reduced test case in CodeSandbox

https://codesandbox.io/s/carbon-overflow-menu-focus-stealing-9k1pl

## Additional information

- Screenshots or code
Here's a picture of the stolen focus after I clicked the Click me button
image

### PR DESCRIPTION
**Pull request by gptt916**
Closes #6227

This PR removes focus button function call from closeMenu, instead only call it when esc key is pressed to close the overflow menu

#### Changelog

**Changed**

- OverflowMenu button no longer aquires focus when the menu is closed via clicking elsewhere on the page.

#### Testing / Reviewing

Leveraging Chrome's focus tracker (https://developers.google.com/web/tools/chrome-devtools/accessibility/focus) verify that the button does not have focus after closing the menu via clicking on the canvas. 

Also verify that closing the menu via the Esc button still results in the menu button aquiring focus.

### CODE REVIEW

File: packages/react/src/components/OverflowMenu/OverflowMenu.js [hunk 1]
```diff
@@ -321,11 +327,11 @@ class OverflowMenu extends Component {
     }
   };
 
-  closeMenu = () => {
-    let wasOpen = this.state.open;
+  closeMenu = (cb) => {
```
Comment (emyarod) [inline packages/react/src/components/OverflowMenu/OverflowMenu.js]: is this change also intentional? the `closeMenu` method is internal and I don't see any new callback usage

Comment (gptt916) [reply packages/react/src/components/OverflowMenu/OverflowMenu.js]: Yep, it's being called on line 310 with the callback, moving the call to `focusMenuEl` to be executed only on Escape press, which I believe was the desired behaviour.

Comment (emyarod) [reply packages/react/src/components/OverflowMenu/OverflowMenu.js]: gotcha looks good to me, let's rename the parameter to something more descriptive like `onCloseMenu` or something

**Review by emyarod — APPROVED** [review]

**Review by andreancardona — CHANGES_REQUESTED** [review]
@gptt916 once param is updated to `onCloseMenu` looks good to me!

**Comment by gptt916** [comment]
@andreancardona Changes are ready for review

**Review by joshblack — APPROVED** [review]

**Review by andreancardona — APPROVED** [review]
@gptt916 🎉

**Pull request merged**